generators/entity-server/templates/src/main/java/package/domain/Entity.java.ejs

```diff
@@ -283,15 +293,15 @@ public class <%= asEntity(entityClass) %> implements Serializable {
     @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
                 <%_ }
             }
-        } else if (databaseType === 'mongodb' || databaseType === 'couchbase') {
-            if (databaseType === 'mongodb') { _%>
+        } else if ((databaseType === 'mongodb' || databaseType === 'couchbase')) {
+            if (databaseType === 'mongodb' && !otherEntityEmbedded) { _%>
     @DBRef
         <%_ } _%>
-    @Field("<%= relationshipFieldName %>")
-        <%_ if (databaseType === 'couchbase') { _%>
+    @Field("<%= relationshipFieldNamePlural %>")
```

Comment:
Is it related to your PR?

Reply:
Actually this is not, it is a bug that I discovered while I was testing the feature. If you like I can remove it for this PR and make an other for that.

Reply:
This should be on a different PR.
It should be reviewed by a JPA developer.
And will be easier to get approved.